pkgs/build-support/node/fetch-yarn-deps/default.nix

```diff
@@ -132,22 +138,52 @@ in
             dontInstall = true;
 
             nativeBuildInputs = [
-              prefetch-yarn-deps
+              (
+                {
+                  "1" = prefetch-yarn-deps;
+                  "3" = yarn-berry_3;
+                  "4" = yarn-berry_4;
+                }
+                .${builtins.toString yarnVersion}
+              )
               cacert
             ];
             GIT_SSL_CAINFO = "${cacert}/etc/ssl/certs/ca-bundle.crt";
             NODE_EXTRA_CA_CERTS = "${cacert}/etc/ssl/certs/ca-bundle.crt";
 
-            buildPhase = ''
-              runHook preBuild
+            configurePhase = lib.optionalString (yarnVersion > 1) ''
+              runHook preConfigure
 
+              export HOME="$NIX_BUILD_TOP"
+              export YARN_ENABLE_TELEMETRY=0
               yarnLock=''${yarnLock:=$PWD/yarn.lock}
-              mkdir -p $out
-              (cd $out; prefetch-yarn-deps --verbose --builder $yarnLock)
-
-              runHook postBuild
+              DIR=$(dirname $yarnLock)
+              cp $DIR/yarn.lock $DIR/package.json $HOME/
```

Comment:
Yes I know.
See https://github.com/NixOS/nixpkgs/pull/355053#discussion_r1847112682
But if you read the follow up, this shouldn't be an issue, because we ususally use the `src` of the parent derivation and have the for

Reply:
OK. For context I tried using this pull request to package a Yarn project outside nixpkgs, and ran into this problem.

I guess what you're saying is that we should always be using

```nix
yarnLock = src + "/yarn.lock";
```
instead of

```nix
yarnLock = src/yarn.lock;
```

Which is understandable but seems like an easy mistake to make so I think mentioning it in the documentation or having a check would be nice.

Reply:
> See [#355053 (comment)](https://github.com/NixOS/nixpkgs/pull/355053#discussion_r1847112682)

Sorry, I'm not sure what's going on with GitHub (issues redesign I guess) but this link isn't working for me, neither by clicking nor opening in a new tab.

Reply:
Mmh weird, I'm on mobile. Maybe because of that..

You can have at the tests at how I solved that problem with the store path.

Thanks for testing it!

Reply:
@the-sun-will-rise-tomorrow is correct that the current state in which `yarnLock` is the argument of the function is slightly misleading. Perhaps after all we should change the function signature of every `fetchYarnDeps` variant now? I'm sorry that I don't have time to put an effort in that myself, I'd have loved to help.

Reply:
I'm with you on the last comment. Lets issue a warning for now, so it catches if only the lock file from the nix store is passed and keep the API the same. In another refactor PR, we can then
1. Move it under the package (e.g. `yarn.fetchDeps` see https://github.com/NixOS/nixpkgs/pull/355053#discussion_r1849863680)
2. Use `src` for `yarn` and `yarn-berry` which will illiminate this current issue